plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/GlueExpressionUtil.java

```diff
@@ -147,6 +147,12 @@ static Optional<String> buildGlueExpressionForSingleDomain(String columnName, Do
             return Optional.empty();
         }
 
+        // Glue throws an exception (e.g. input string: "__HIVE_D" is not an integer)
+        // for column <> '__HIVE_DEFAULT_PARTITION__' or column = '__HIVE_DEFAULT_PARTITION__' expression on numeric types
```

Comment:
Is there is any specific reason on not using `NOT` or `IS NULL` logical operator provided by Glue ? It was mentioned here - but any reason on not using it https://docs.aws.amazon.com/glue/latest/dg/aws-glue-api-catalog-partitions.html#aws-glue-api-catalog-partitions-GetPartitions

Reply:
@findinpath and I confirmed the expression returns incorrect result. Let me leave a code comment.

Reply:
@Praveen2112 Added a comment. Please take another look.